### DISCUSSION TITLE
About installation error on raspberray pi3

### DISCUSSION THREAD

**Discussion by akiracc** [dls-controls/pythonSoftIOC#120] [General]
I have installed softioc on both Raspberry Pi4 and Pi3 using pip. The installation was successful on Pi4, but encountered an error on Pi3. Both Pi4 and Pi3 are running the same 64-bit OS version (Debian 11 bullseye) and have the same software package installed.

However, I am facing difficulty in understanding why Pi3 is unable to install pythonSoftIOC. I am seeking help to resolve this issue. Could anyone kindly suggest a solution? Here's the error message received:
```
[Error message]
====================================================================
   error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 4
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for epicscorelibs
Failed to build epicscorelibs
ERROR: Could not build wheels for epicscorelibs, which is required to install pyproject.toml-based projects
====================================================================
```
By the way, I noticed that Pi3 takes an unusually long time to compile softioc-4.2.0.tar.gz and epicscorelibs-7.0.7.99.0.0.tar.gz compared to Pi4. In my opinion, the time taken is longer than what their abilities would suggest.

**Comment by AlexanderWells-diamond** [comment]
Is that the complete error message? I would have expected to see some more in-depth error of exactly where the build failed. If there is any more please provide it.

You may also be able to get more information by directly installing epicscorelibs, either directly: `pip install epicscorelibs` or by skipping any pre-built wheels: `pip install --no-binary :all: epicscorelibs` (although I don't think there are any for ARM so that is probably unnecessary).

Ultimately this is probably not an issue with pythonSoftIoc itself, but rather an issue on [its dependency](https://github.com/mdavidsaver/epicscorelibs). We can do a little investigation from our end (and also investigate running our builds on ARM) but any solution will likely require changes elsewhere.
